File: examples/java/getting-started/quickstart.java
import java.time.Duration;
import java.util.Map;
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;

public class QuickstartFunction extends DurableHandler<Map<String, String>, Map<String, Object>> {

    @Override
    public Map<String, Object> handleRequest(Map<String, String> event, DurableContext context) {
        String message = context.step("step-1", String.class, stepCtx -> {
            stepCtx.getLogger().info("Hello from step-1");
            return "Hello from Durable Lambda!";
        });

        // Pause for 10 seconds without consuming CPU or incurring usage charges
        context.wait("wait-10s", Duration.ofSeconds(10));

        // Replay-aware: logs once even though the function replays after the wait
        context.getLogger().info("Resumed after wait");

        return Map.of("statusCode", 200, "body", message);
    }
}

File: examples/csharp/getting-started/quickstart.cs
using Amazon.Lambda.Core;
using Amazon.Lambda.DurableExecution;
using Microsoft.Extensions.Logging;

public class QuickstartFunction
{
    public Task<DurableExecutionInvocationOutput> Handler(
        DurableExecutionInvocationInput input, ILambdaContext context)
        => DurableFunction.WrapAsync<object, Response>(Workflow, input, context);

    private async Task<Response> Workflow(object input, IDurableContext ctx)
    {
        string message = await ctx.StepAsync(
            async (stepCtx, _) =>
            {
                stepCtx.Logger.LogInformation("Hello from step-1");
                return "Hello from Durable Lambda!";
            },
            name: "step-1");

        // Pause for 10 seconds without consuming CPU or incurring usage charges
        await ctx.WaitAsync(TimeSpan.FromSeconds(10), name: "wait-10s");

        // Replay-aware: logs once even though the function replays after the wait
        ctx.Logger.LogInformation("Resumed after wait");

        return new Response(200, message);
    }
}

public record Response(int StatusCode, string Body);
